core/src/main/java/nl/nn/adapterframework/jdbc/JdbcTransactionalStorage.java

```diff
@@ -779,23 +778,21 @@ public String storeMessage(Connection conn, String messageId, String correlation
 		}
 	}
 
-
-
-
-	private S retrieveObject(ResultSet rs, int columnIndex, boolean compressed) throws ClassNotFoundException, JdbcException, IOException, SQLException {
+	private RawMessageWrapper<S> retrieveObject(ResultSet rs, int columnIndex, boolean compressed) throws ClassNotFoundException, JdbcException, IOException, SQLException {
 		try (InputStream blobInputStream = JdbcUtil.getBlobInputStream(getDbmsSupport(), rs, columnIndex, compressed)) {
 			if (blobInputStream==null) {
 				return null;
 			}
 			try (ObjectInputStream ois = new ObjectInputStream(blobInputStream)) {
-				return (S)ois.readObject();
+				RawMessageWrapper<S> messageWrapper = new RawMessageWrapper<>((S)ois.readObject());
```

Comment:
ois.readObject() instanceof MessageWrapper, dan niet nog een keer wrappen.

Reply:
MessageWrapper is een subclass van RawMessageWrapper, dus dat gaat goed.